framework/src/outputs/AdvancedOutput.C

```diff
@@ -429,13 +429,19 @@ AdvancedOutput::initAvailableLists()
         if (var.isArray())
           vname = SubProblem::arrayVariableComponent(var_name, i);
 
-        if (type.order == CONSTANT && type.family != MONOMIAL_VEC)
+        // A note that if we have p-refinement we assume "worst-case" scenario that our constant
+        // monomial/monomial-vec families have been refined and we can no longer write them as
+        // elemental
```

Comment:
this is making me think of transfers

both nearest-location and shape evaluation have special logic for constant monomials

we need to be returning the current order of the family, not the initial one, to avoid fooling downstream code

Reply:
whole auxkernel system has a special treatment for const monomial iirc

Reply:
I've opened https://github.com/idaholab/moose/issues/25465. The aux kernel system checks the number of dofs so it's robust

Reply:
sounds good